docs/content/en/docs/soarca-extensions/fin-python-library.md

```diff
@@ -0,0 +1,210 @@
+        name="pong_output",
+        description="If ping, return pong",
+        value="pong",
+        constant=True,
+        external=False)
+
+    context = command.command.context
+
+    return ResultStructure(
+        state="success", context=context, variables={"result": result})
+
+
+def main(mqtt_broker: str, mqtt_port: int, username: str, password: str) -> None:
+
+    finId = "soarca-fin--pingpong-f877bb3a-bb37-429e-8ece-2d4286cf326d"
+    agentName = "soarca-fin-pong-f896bb3b-bb37-429e-8ece-2d4286cf326d"
+    externalReferenceName = "external-reference-example-name"
+    capabilityId = "mod-pong--e896aa3b-bb37-429e-8ece-2d4286cf326d"
+
+    # Create AgentStructure
+    agent = AgentStructure(
+        name=agentName)
+
+    # Create ExternalReference
+    external_reference = ExternalReference(name=externalReferenceName)
+
+    # Create StepStructure
+    step_structure = StepStructure(
+        name="step_name",
+        description="step description",
+        external_references=[external_reference],
+        command="pong",
+        target=agentName)
+
+    # Create CapabilityStructure
+    capability_structure = CapabilityStructure(
+        capability_id=capabilityId,
+        type=WorkFlowStepEnum.action,
+        name="Ping Pong capability",
+        version="0.0.1",
+        step={
+            "test": step_structure},
+        agent={
+            "testagent": agent})
+
+    # Create Soarca fin
+    fin = SoarcaFin(finId)
+    # Set config for MQTT Server
+    fin.set_config_MQTT_server(mqtt_broker, mqtt_port, username, password)
+    # Register Capabilities
+    fin.create_fin_capability(capability_structure, capability_pong_callback)
+    # Start the fin
+    fin.start_fin()
+
+
+if __name__ == "__main__":
+    load_dotenv()
+    MQTT_BROKER = os.getenv("MQTT_BROKER", "localhost")
+    MQTT_PORT = int(os.getenv("MQTT_PORT", "1883"))
+    USERNAME = os.getenv("MQTT_USERNAME", "soarca")
+    PASSWD = os.getenv("MQTT_PASSWD", "password")
+
+    main(MQTT_BROKER, MQTT_PORT, USERNAME, PASSWD)
+```
+
+## Architecture
+### Application Layout
+The main object of the application is the `SoarcaFin` object, which is responsible for configuring and creating and controlling the capabilities.
+The SoarcaFin creates `MQTTClient`s for each capability registered, plus one for registering, unregistering and controlling the fin.
+`MQTTClient`s each have their own connection to the MQTT Broker and own `Parser` and `Executor` objects.
+The `Parser` object parsers the raw MQTT messages and tries to convert them to one of the objects in `src/models`.
+The `Executor` runs in their own thread and handles the actual execution of the messages.
+The `Executor` polls a thread-safe queue for new messages and performs IO operations, such as sending messages to the MQTT broker and calling capability callbacks.
+
+### Setup SOARCA Capabilities
+To register a fin to SOARCA, first create a `SoarcaFin` object and pass the `fin_id` in the constructor.
```

Comment:
Are the specific requirements to the `fin_id`? should this be in a certain format? please explain